### PR TITLE
Added Request Specs

### PR DESCRIPTION
**Pull request by BryteLitty**
In this PR, I added request specs to test each controller for the following conditions:

- if the response status was correct
- if a correct template was rendered
- if the response body includes the correct placeholder text

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+
+    it 'renders the correct index page' do
+      get '/users/1/posts'
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes the correct placeholder text' do
+      get '/users/1/posts'
+      expect(response.body).to include('Here is a list of posts for a given user 1')
+    end
```
Comment (Qoosim) [inline spec/requests/posts_spec.rb]: - [ ] Great job so far. However, you're expected to write tests for post `index` template and the placeholder and `show` template and placeholder. You only wrote for the index not show. Kindly complete them.

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+      get '/users/1'
+      expect(response).to render_template(:show)
+    end
+
+    it 'includes the correct placeholder text' do
+      get '/users/1'
+      expect(response.body).to include('Users ID: 1')
+    end
   end
 end
```
Comment (Qoosim) [inline spec/requests/users_spec.rb]: - [ ] Same thing happened here too. You only wrote tests for the user `show` template and placeholder. It remains for the `index` template and placeholder.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @BryteLitty, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: @BryteLitty, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Test units for both index and show templates have been added as per requirements :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**